Makefile

```diff
@@ -90,9 +91,11 @@ ALWAYS:
 
 # build our fork of libsodium, placing artifacts into crypto/lib/ and crypto/include/
 crypto/lib/libsodium.a:
-	cd crypto/libsodium-fork && \
-		./autogen.sh && \
-		./configure --disable-shared --prefix="$(SRCPATH)/crypto/" && \
+	mkdir -p crypto/copies/$(OS_TYPE)/$(ARCH)
+	cp -R crypto/libsodium-fork crypto/copies/$(OS_TYPE)/$(ARCH)/libsodium-fork
```

Comment:
A more common pattern is to check in the results of running `autogen.sh` into git, and then do builds by invoking `configure` from a different directory.  The `configure` script takes care of setting up symlinks, etc.

In particular, I'm thinking of something like:

```
crypto/libs/$(OS_TYPE)/$(ARCH)/Makefile:
  mkdir -p $(@D)
  cd $(@D) && ../../../libsodium-fork/configure --disable-shared --prefix="$(SRCPATH)/$(@D)"

crypto/libs/$(OS_TYPE)/$(ARCH)/src/libsodium/.libs/libsodium.a: crypto/libs/$(OS_TYPE)/$(ARCH)/Makefile
  cd crypto/libs/$(OS_TYPE)/$(ARCH) && make

crypto/libs/$(OS_TYPE)/$(ARCH)/libsodium.a: crypto/libs/$(OS_TYPE)/$(ARCH)/src/libsodium/.libs/libsodium.a
  cd $(@D) && make install
```

Reply:
I've never used automake before, does this not produce files that are specific to the platform it's running on?

Reply:
The results of autogen.sh should be platform-agnostic

Reply:
+1 to Nikolai, it is common to generate platform-dependant build scripts to a dir and run the build from there without copying sources.

Reply:
I've been thinking and this effort seems out of scope for the story I'm working on since I would need to implement a new feature that doesn't exist in our libsodium-fork. Also I think moving diverging from upstream libsodium's build automation could be a little dangerous for us over time.

Reply:
What feature are you referring to?

Reply:
The feature of supporting multi-platform concurrent builds of libsodium using a shared working directory.

Reply:
I agree though that's something we would want if we don't have to diverge too much from upstream to implement it, I'm just not sure if that's something within the scope of this work effort

Reply:
It's built into libsodium as-is.

Reply:
I can't seem to figure out how to do that, since automake and configure seem to be making a lot of files for the host operating system in the current working directory. Would you be alright with me making an issue to target this feature in a future sprint?

Reply:
Fine by me.

automake and configure are very different in these respects.  automake is platform-agnostic.  configure is what populates the build directory with platform-specific files.